### ISSUE TITLE
Rewrite color Sass maps to enable theming

### ISSUE THREAD

**Issue by maya** [uswds/uswds#2396]
Rewrite the color Sass maps like describe here: https://alligator.io/sass/sass-maps/

For example:
```
$red: (
  5: #fcf5f5,
  10: #f7ddda,
  20: #f7b7ad,
  30: #f28f88,
  40: #ed6b61,
  50: #d83731,
  60: #ab3a3a,
  70: #6e312f,
  80: #3b2523,
  90: #211b1b
  …
);

$blue: (
...
  30: #79bcf2,
  40: #56a4e3,
  50: #317cb9,
  60: #2f6794
…
);
```

This would enable:
```
$primary: $red;

.bg-accent-light {
  background-color: map-get($primary, 10);
}
```
which would output:
```
.bg-accent-light {
  background-color: $red-40;
}
```

and could be used for any color 🎉

**Comment by thisisdano** [comment]
Jumping on this now

**Comment by maya** [comment]
Closed via #2399. For the future, we need to think about how to enable vivid variants in themes.
